src/modules/paymentProcessor/bridging/abstracts/CrosschainBase_v1.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+import {IOrchestrator_v1} from
+    "src/orchestrator/interfaces/IOrchestrator_v1.sol";
+import {IPaymentProcessor_v1} from "@pp/IPaymentProcessor_v1.sol";
+import {IERC20PaymentClientBase_v1} from
+    "@lm/interfaces/IERC20PaymentClientBase_v1.sol";
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+//import {IPP_CrossChain_v1} from "./IPP_Template_v1.sol";
+import {ERC165Upgradeable, Module_v1} from "src/modules/base/Module_v1.sol";
+import {ICrossChainBase_v1} from "./ICrosschainBase_v1.sol";
+/**
+ * @title   Inverter Template Payment Processor
+ *
+ * @notice  Basic template payment processor used as base for developing new payment processors.
+ *
+ * @dev     This contract is used to showcase a basic setup for a payment processor. The contract showcases the
+ *          following:
+ *          - Inherit from the Module_v1 contract to enable interaction with the Inverter workflow.
+ *          - Use of the IPaymentProcessor_v1 interface to facilitate interaction with a payment client.
+ *          - Implement custom interface which has all the public facing functions, errors, events and structs.
+ *          - Pre-defined layout for all contract functions, modifiers, state variables etc.
+ *          - Use of the ERC165Upgradeable contract to check for interface support.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+
+contract CrosschainBase_v1 is ICrossChainBase_v1, Module_v1 {
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId_ == type(ICrossChainBase_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+    //--------------------------------------------------------------------------
+    // State
+
+    /// @dev Mapping of payment ID to bridge transfer return data
+    mapping(uint => bytes) internal _bridgeData;
```

Comment:
How do we make sure that the implementation contract, inheriting from the base, is using this `bytes` storage? Not sure about this, but we could make it private and enforce a setter for it (maybe). Similar as what is done with ERC20 abstracts and the token balance

Reply:
Also curious if you guys looked into common return values from bridge protocol to conclude this to be the right data to be stored.

Reply:
Different bridges return different types of data:
- Hyperlane returns bytes32 (messageId)
- Connext returns uint (intentID)
- LayerZero returns nothing

The `_bridgeData` mapping isn't actually necessary because:
1. The return values are primarily used for tracking/monitoring on the source chain
2. Each bridge implementation can handle its return data differently based on its specific needs
3. The destination chain execution isn't dependent on these return values

Regarding enforcing usage through a setter - since the return data isn't critical for cross-chain functionality, we can leave it up to the implementing contracts to decide if and how they want to handle bridge return data. This gives more flexibility to handle the varying return types from different bridge protocols. So for now I'm thinking we remove this.

Reply:
Fixed in new commit